src/app/clusters/commissioner-control-server/commissioner-control-server.h

```diff
@@ -0,0 +1,180 @@
+};
+
+struct CommissionNodeInfo
+{
+    CommissioningWindowParams params;
+    Optional<ByteSpan> ipAddress;
+    Optional<uint16_t> port;
+    uint8_t ipAddressBuffer[kIpAddressBufferSize];
+
+    CHIP_ERROR SetIPAddress(const Optional<ByteSpan> & address)
+    {
+        if (!address.HasValue())
+        {
+            ipAddress.ClearValue();
+            return CHIP_NO_ERROR;
+        }
+
+        const ByteSpan & addressSpan = address.Value();
+        size_t addressLength         = addressSpan.size();
+
+        if (addressLength > kIpAddressBufferSize)
+        {
+            return CHIP_ERROR_BUFFER_TOO_SMALL;
+        }
+
+        if (addressLength == 0)
+        {
+            ipAddress.ClearValue();
+            return CHIP_NO_ERROR;
+        }
+
+        memcpy(ipAddressBuffer, addressSpan.data(), addressLength);
+        ipAddress.SetValue(ByteSpan(ipAddressBuffer, addressLength));
+        return CHIP_NO_ERROR;
+    }
+};
+
+class Delegate
+{
+public:
+    /**
+     * @brief Handle a commissioning approval request.
+     *
+     * This command is sent by a client to request approval for a future CommissionNode call.
+     * The server SHALL always return SUCCESS to a correctly formatted RequestCommissioningApproval
+     * command, and then send a CommissioningRequestResult event once the result is ready.
+     *
+     * @param request The commissioning approval request to handle.
+     * @return CHIP_ERROR indicating the success or failure of the operation.
+     */
+    virtual CHIP_ERROR HandleCommissioningApprovalRequest(const CommissioningApprovalRequest & request) = 0;
+
+    /**
+     * @brief Validate a commission node command.
+     *
+     * This command is sent by a client to request that the server begins commissioning a previously
+     * approved request.
+     *
+     * The server SHALL return FAILURE if the CommissionNode command is not sent from the same
+     * NodeId as the RequestCommissioningApproval or if the provided RequestId to CommissionNode
+     * does not match the value provided to RequestCommissioningApproval.
+     *
+     * The validation SHALL fail if the client Node ID is kUndefinedNodeId, such as getting the NodeID from
+     * a group or PASE session.
+     *
+     * @param clientNodeId The NodeId of the client.
+     * @param requestId The request ID to validate.
+     * @return CHIP_ERROR indicating the success or failure of the operation.
+     */
+    virtual CHIP_ERROR ValidateCommissionNodeCommand(NodeId clientNodeId, uint64_t requestId) = 0;
+
+    /**
+     * @brief Get the parameters for the commissioning window.
+     *
+     * This method is called to retrieve the parameters needed for the commissioning window.
+     *
+     * @param[out] outParams The parameters for the commissioning window.
+     * @return CHIP_ERROR indicating the success or failure of the operation.
+     */
+    virtual CHIP_ERROR GetCommissioningWindowParams(CommissioningWindowParams & outParams) = 0;
```

Comment:
Where is this used?

Reply:
Removed

Reply:
Yufeng pinged me and mentioned section of code where it was supposed to be used but was accidentally removed. It is now added back in